dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -576,6 +595,28 @@ internal open class RumViewScope(
         return stopped && activeResourceScopes.isEmpty() && (pending <= 0L)
     }
 
+    /*
+     * The refresh rate needs to be computed with each view because:
+     * - it requires a context with a UI (we can't get this from the application context);
+     * - it can change between different activities (based on window configuration)
+     */
+    @Suppress("DEPRECATION")
+    private fun detectRefreshRateScale(key: Any) {
+        val activity = when (key) {
+            is Activity -> key
+            is Fragment -> key.activity
+            is android.app.Fragment -> key.activity
+            else -> null
+        } ?: return
+
+        val display = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            activity.display
+        } else {
+            (activity.getSystemService(Context.WINDOW_SERVICE) as? WindowManager)?.defaultDisplay
+        } ?: return
+        refreshRateScale = 60.0 / display.refreshRate
```

Comment:
> The display refresh rate can't change during an Activity's lifecycle

I didn't find the docs which will say that. It seems that on refresh rate there is no config change per https://developer.android.com/guide/topics/manifest/activity-element#config and because docs here https://developer.android.com/guide/topics/media/frame-rate propose to use https://developer.android.com/reference/android/hardware/display/DisplayManager#registerDisplayListener(android.hardware.display.DisplayManager.DisplayListener,%20android.os.Handler) to listen for it.

But again, maybe I'm wrong - I have no device to test such things anyway.

Reply:
Maybe we could move this magic number in a static constant